### ISSUE TITLE
OJS 3.3 Release Candidate: Public file in publisher library not available

### ISSUE THREAD

**Issue by kshuttle** [pkp/pkp-lib#6558]
This issue came up in testing for the 3.3 release candidate. After uploading a file to the publisher library and making it public, the file is **not** accessible by an anonymous user via direct URL. This was tested in the[ Journal of PKP: OJS Testing](https://sfulib710.publicknowledgeproject.org/ojs3/testing-3_3/index.php/ojs33_testing/index). Under Workflow > Publisher Library, the following file was uploaded and URL provided:
![image](https://user-images.githubusercontent.com/59741486/104073651-15bb4880-51c3-11eb-91e9-6d7d50650be1.png)
The URL goes to a 403 Forbidden page when tested: https://sfulib710.publicknowledgeproject.org/ojs3/testing-3_3/index.php/ojs33_testing/libraryFiles/downloadPublic/id

**Comment by asmecher** [comment]
@Vitaliy-1, could you take a look at this? (If you've already fixed related aspects since the 3.3rc2 release candidate, this may have already been resolved.)

**Comment by Vitaliy-1** [comment]
I'm unable to reproduce it with the current master branch OJS/OMP/OPS. 
Strangely, I can download a file by the [URL](https://sfulib710.publicknowledgeproject.org/ojs3/testing-3_3/index.php/ojs33_testing/libraryFiles/downloadPublic/id/1) specified, tried several scenarios but maybe I'm missing something. Is this file still inaccessible?

**Comment by NateWr** [comment]
I get a 403 response with the URL you posted, @Vitaliy-1. Were you by any chance logged in to that testing install?

**Comment by Vitaliy-1** [comment]
Yep, now it's 403 for me either. Initially, I'd tried several scenarios, including Tor. I don't have access to that instance, thus all attempts were without being logged in.

**Comment by NateWr** [comment]
OK I figured out what the issue is. When first uploading a file, the form shows the URL with a placeholder, `id`, instead of the ID of the file:

```
https://sfulib710.publicknowledgeproject.org/ojs3/testing-3_3/index.php/ojs33_testing/libraryFiles/downloadPublic/id
```

After upload, it replaces that with the correct ID of the file:

```
https://sfulib710.publicknowledgeproject.org/ojs3/testing-3_3/index.php/ojs33_testing/libraryFiles/downloadPublic/3
```

So the access control is working, but the upload form is confusing. It shouldn't show the URL until after the file is created and an ID exists. I'll update the original post to reflect this.

**Comment by NateWr** [comment]
@Vitaliy-1 I've unassigned you from this because I think there are more important bugs to fix first for 3.3.

**Comment by NateWr** [comment]
I'm going to remove this from the 3.3 milestone as we're running out of time and this isn't a critical issue.
